Cannot use octopusCreateRelease releaseNotesFile - java.io.NotSerializableException: hudson.plugins.octopusdeploy.Log

Plugin version 4.1.1, Jenkins Version 2.528.3

Using releaseNotesSource 'scm' works fine. 'file' throws a FATAL error.

We were running a year old plugin version, but we updated to 4.1.1 and restarted Jenkins. The error persists.

Pipeline step:

           steps {
                script {
					
		   unstash 'notes' // previously stashed in a prior stage on  a linux agent, this step is running on a windows agent
                  
                    def content = readFile(file: 'release-notes.md', encoding: 'UTF-8')
                    echo "Content read: ${content}"  // displays content as expected
                    					
                    octopusCreateRelease (
                        serverId: 'octopus-nz-geois',
                        spaceId: 'Spaces-1',
                        project: "${env.OCTO_PROJECT}",
                        channel: "${env.OCTO_CHANNEL}",
                        releaseVersion: "${env.APP_VERSION}",
                        defaultPackageVersion: "${env.APP_VERSION}",
                        deploymentTimeout: '00:10:00',
                        cancelOnTimeout: true,
                        jenkinsUrlLinkback: true,
                        releaseNotes: true,
                        releaseNotesSource: 'file',
                        releaseNotesFile: 'release-notes.md',
                        toolId: 'Chocolatey Octo'
                    )
                }
            }

Console output

[Pipeline] octopusCreateRelease
INFO: Started Octopus Release
INFO: =======================
INFO: Project: BusSys - AppMessages v4
INFO: Release Version: 1.0.0.29
INFO: Channel: Default
INFO: Include Release Notes?: true
INFO: 	Release Notes Source: file
INFO: 	Release Notes File: release-notes.md
INFO: Deploy this Release?: false
INFO: Package Configurations: none
INFO: =======================
FATAL: Unable to get file contents from release notes file! - Unable to serialize hudson.plugins.octopusdeploy.OctopusDeployReleaseRecorder$ReadFileCallable@952727d
java.io.IOException: Unable to serialize hudson.plugins.octopusdeploy.OctopusDeployReleaseRecorder$ReadFileCallable@952727d
	at hudson.remoting.UserRequest.serialize(UserRequest.java:294)
	at hudson.remoting.UserRequest.<init>(UserRequest.java:103)
	at hudson.remoting.Channel.call(Channel.java:1106)
	at hudson.FilePath.act(FilePath.java:1204)
	at hudson.FilePath.act(FilePath.java:1193)
	at PluginClassLoader for octopusdeploy//hudson.plugins.octopusdeploy.OctopusDeployReleaseRecorder.getReleaseNotesFromFile(OctopusDeployReleaseRecorder.java:484)
	at PluginClassLoader for octopusdeploy//hudson.plugins.octopusdeploy.OctopusDeployReleaseRecorder.perform(OctopusDeployReleaseRecorder.java:296)
	at jenkins.tasks.SimpleBuildStep.perform(SimpleBuildStep.java:123)
	at PluginClassLoader for workflow-basic-steps//org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:101)
	at PluginClassLoader for workflow-basic-steps//org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:71)
	at PluginClassLoader for workflow-step-api//org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution.lambda$start$0(SynchronousNonBlockingStepExecution.java:49)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.io.NotSerializableException: hudson.plugins.octopusdeploy.Log
	at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1200)
	at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1585)
	at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1542)
	at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
	at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
	at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1585)
	at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1542)
	at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
	at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
	at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:358)
	at hudson.remoting.UserRequest._serialize(UserRequest.java:283)
	at hudson.remoting.UserRequest.serialize(UserRequest.java:292)
	... 15 more`

--- a/src/main/java/hudson/plugins/octopusdeploy/OctopusDeployReleaseRecorder.java
+++ b/src/main/java/hudson/plugins/octopusdeploy/OctopusDeployReleaseRecorder.java
@@ -295,7 +295,7 @@ public class OctopusDeployReleaseRecorder extends AbstractOctopusDeployRecorderP
                 try {
                     releaseNotesContent += getReleaseNotesFromFile(workspace, releaseNotesFile);
                 } catch (Exception ex) {
-                    log.fatal(String.format("Unable to get file contents from release notes file! - %s", getExceptionMessage(ex)));
+                    log.fatal(String.format("Unable to get file contents from release notes file '%s'! - %s", releaseNotesFile, getExceptionMessage(ex)));
                     success = false;
                 }
             } else if (isReleaseNotesSourceScm()) {
@@ -475,6 +475,7 @@ public class OctopusDeployReleaseRecorder extends AbstractOctopusDeployRecorderP
     /**
      * Return the release notes contents from a file.
      * @param workspace our build
+     * @param releaseNotesFilename the name of the file that contains the release notes
      * @return string contents of file
      * @throws IOException if there was a file read io problem
      * @throws InterruptedException if the action for reading was interrupted

--- a/src/main/java/hudson/plugins/octopusdeploy/OctopusDeployReleaseRecorder.java
+++ b/src/main/java/hudson/plugins/octopusdeploy/OctopusDeployReleaseRecorder.java
@@ -293,7 +293,7 @@ public class OctopusDeployReleaseRecorder extends AbstractOctopusDeployRecorderP
         if (releaseNotes) {
             if (isReleaseNotesSourceFile()) {
                 try {
-                    releaseNotesContent += getReleaseNotesFromFile(workspace, releaseNotesFile, log);
+                    releaseNotesContent += getReleaseNotesFromFile(workspace, releaseNotesFile);
                 } catch (Exception ex) {
                     log.fatal(String.format("Unable to get file contents from release notes file! - %s", getExceptionMessage(ex)));
                     success = false;
@@ -479,32 +479,18 @@ public class OctopusDeployReleaseRecorder extends AbstractOctopusDeployRecorderP
      * @throws IOException if there was a file read io problem
      * @throws InterruptedException if the action for reading was interrupted
      */
-    private String getReleaseNotesFromFile(FilePath workspace, String releaseNotesFilename, Log log) throws IOException, InterruptedException {
+    private String getReleaseNotesFromFile(FilePath workspace, String releaseNotesFilename) throws IOException, InterruptedException {
         FilePath path = new FilePath(workspace, releaseNotesFilename);
-        return path.act(new ReadFileCallable(log));
+        return path.act(new ReadFileCallable());
     }
 
     /**
      * This callable allows us to read files from other nodes - ie. Jenkins slaves.
      */
     private static final class ReadFileCallable implements FileCallable<String> {
-        public final static String ERROR_READING = "<Error Reading File>";
-
-        private final Log log;
-
-        public ReadFileCallable(Log log)
-        {
-            this.log = log;
-        }
-
         @Override
-        public String invoke(File f, VirtualChannel channel) {
-            try {
-                return StringUtils.join(Files.readAllLines(f.toPath(), StandardCharsets.UTF_8), "\n");
-            } catch (IOException ex) {
-                log.error("Failed to read file: " + getExceptionMessage(ex));
-                return ERROR_READING;
-            }
+        public String invoke(File f, VirtualChannel channel) throws IOException {
+            return StringUtils.join(Files.readAllLines(f.toPath(), StandardCharsets.UTF_8), "\n");
         }
 
         @Override

